index.d.ts

```diff
@@ -0,0 +1,20 @@
+import { GraphQLFieldResolver, GraphQLResolveInfo  } from 'graphql';
+
+type RawConfig = {};
+type PreparedConfig = { root: object, pick: object };
+type Projection = object | undefined;
+type ProjectMethod = (info: GraphQLResolveInfo) => Projection;
+type Resolver = GraphQLFieldResolver<any,any>;
+
+type gqlProjection = (config: RawConfig) => { project: ProjectMethod, resolvers: Resolver };
+type prepareConfig = (config: RawConfig) => PreparedConfig;
+type genProjection = (config: PreparedConfig) => ProjectMethod;
+type genResolvers = (config: PreparedConfig) => Resolver[]
```

Comment:
`genResolvers` does NOT give `GraphQLFieldResolver<,>`.
It produces:
```javascript
{
  User: {
    userId: (parent, args, context, info) => parent._id,
    field1: (parent, args, context, info) => parent.mongoA,
  },
}
```
while `GraphQLFieldResolver<,>` is essentially the type of `(parent, args, context, info) => any`.

Reply:
Indeed you're right. I've missed a level.